### PR TITLE
CEDS-3456 Fix tabnabbing

### PR DESCRIPTION
**Pull request by MaciejRewera**

### CODE REVIEW

**Review by glidester — APPROVED** [review]
LGTM

**Comment by udinobi** [comment]
`detailsContentLink` template also has `target="_blank"`

File: app/views/components/gds/link.scala.html [hunk 1]
```diff
@@ -27,7 +27,7 @@
 
 @hasHint = @{textHidden.exists(_.nonEmpty)}
 
-<a class="@classes" href="@call" target=@target @id.map{id => id="@id" }>
- <span @if(hasHint){aria-hidden="true"}>@text</span>
- @if(hasHint){<span class="govuk-visually-hidden">@{textHidden.getOrElse("")}</span>}
-</a>
```
Comment (glidester) [inline app/views/components/gds/link.scala.html:33]: Maybe this new extra line is the cause of the space?

File: app/views/components/gds/tariffLink.scala.html [hunk 2]
```diff
@@ -14,6 +14,14 @@
  * limitations under the License.
  *@
 
+@import views.html.components.gds.link
+
+@this(link: link)
+
 @(linkText: String, linkUrl: String)
 
-<a class="govuk-link govuk-link--no-visited-state" target="_blank" href="@linkUrl">@linkText</a>
```
Comment (glidester) [inline app/views/components/gds/tariffLink.scala.html:19]: ... and here too

File: app/views/declaration/not_declarant.scala.html [hunk 3]
```diff
@@ -23,15 +23,18 @@
...
 )
 
 @()(implicit request: Request[_], messages: Messages)
 
 @paragraph1Link = {<a href=@controllers.routes.RootController.displayPage.url>@messages("notDeclarent.paragraph1Link")</a>}
 @paragraph2Link = {<a href="https://www.gov.uk/guidance/dispatching-your-goods-within-the-eu">@messages("notDeclarent.paragraph2Link")</a>}
 @supportPhone = {<b>0300 200 3700</b>}
-@enquiriesLink = {<a href="https://www.gov.uk/government/organisations/hm-revenue-customs/contact/customs-international-trade-and-excise-enquiries" target="_blank" rel="noopener noreferrer">@messages("general.inquiries.help.link")</a>}
+@enquiriesLink = {
+  @link(text = messages("general.inquiries.help.link"), call = Call("GET", "https://www.gov.uk/government/organisations/hm-revenue-customs/contact/customs-international-trade-and-excise-enquiries"), target = "_blank")
```
Comment (udinobi) [inline app/views/declaration/not_declarant.scala.html:36]: It's not that I'm against to have the url hardcoded here, but it seems to me at odds with the logic we usually follow... the urls are in `application.conf`. And this is the same for all the other urls I'm seeing hardcoded in the templates in this PR (I do not want to repeat the same comment)

**Review by udinobi — APPROVED** [review]

File: app/views/components/gds/link.scala.html [hunk 4]
```diff
@@ -27,7 +27,5 @@
...
\ No newline at end of file
+<a class="@classes" href="@call" target=@target @id.map{id => id="@id" } @if(target == "_blank"){rel="noopener noreferrer"}
```
Comment (glidester) [inline app/views/components/gds/link.scala.html:30]: Would be wise to add a comment here informing future devs not to split the `<a>` tag into multiple lines. The effect on messages with links is far from obvious so its likely they would not realise (and the problem will then return) :zombie:

Comment (udinobi) [reply app/views/components/gds/link.scala.html:30]: :muscle:

Comment (MaciejRewera) [reply app/views/components/gds/link.scala.html:30]: I'll add that, but if you try to format this piece of code, then you'd have a few tests failing. So you'll definitely notice ;-)

**Review by glidester — APPROVED** [review]

**Pull request merged**